app/views/helpers/injected/obligations/NextUpdatesHelperR17.scala.html

```diff
@@ -0,0 +1,209 @@
+        @link(link = getMessage("quarterly.compatibleSoftware.link"), target = Some("_blank"), messageKey = getMessage("quarterly.ny.text2"), id = Some("quarterly-compatible-software-link-ny"))
+    }
+    @p(id = Some("next-year-desc2")){ @getMessage("quarterly.ny.text3") }
+}
+
+@nextYearTabContentAnnual = {
+    @h2(getMessage("annual.ny.heading", currentTaxYearEnd, nextTaxYearEnd), classes = "govuk-heading-l", optId = Some("next-year-heading"))
+    @p(id = Some("next-year-desc")){
+        @getMessage("annual.text2")
+        @link(link = getMessage("quarterly.compatibleSoftware.link"), target = Some("_blank"), messageKey = getMessage("annual.link"), id = Some("annual-compatible-software-link-ny"))
+    }
+    @p(id = Some("next-year-desc2")){ @getMessage("annual.ny.text1") }
+}
+
+@quarterlyDropdownContent = {
+    @p(id = Some("current-year-dropdown-desc")){@getMessage("quarterly.dropdown.text1")}
+    @bulletList(
+        itemMsgKeys =
+            getMessage("quarterly.dropdown.text2"),
+            getMessage("quarterly.dropdown.text3"),
+    )
+    @p(id = Some("current-year-dropdown-desc2")) {
+        @getMessage("quarterly.dropdown.text4")
+        @link(link = getMessage("quarterly.compatibleSoftware.link"), target = Some("_blank"), messageKey = getMessage("quarterly.dropdown.link.text"), id = Some("quarterly-compatible-software-link-dropdown"))
+    }
+}
+
+@quarterlyUpdatesTable = {
+    @govukTable(Table(
+        head = Some(List(
+            HeadCell(
+                content = Text(getMessage("quarterly.table.heading.row1")),
+                attributes = Map("id" -> "table-head-name-deadline")),
+            HeadCell(
+                content = Text(getMessage("quarterly.table.heading.row2")),
+                attributes = Map("id" -> "table-head-name-period")),
+            HeadCell(
+                content = Text(getMessage("quarterly.table.heading.row3")),
+                attributes = Map("id" -> "table-head-name-updates-due"))
+        )),
+        rows = quarterlyUpdatesRows,
+        attributes = Map("id" -> "quarterly-updates-table"),
+    ))
+}
+
+@quarterlyUpdatesRows = @{
+    viewModel.allDeadlines.zipWithIndex.map { case (DeadlineViewModel(obligationType: ObligationType, bothQuarters: Boolean, deadlineDate: LocalDate, standardObligations: Seq[ObligationWithIncomeType], calendarObligations: Seq[ObligationWithIncomeType]), index) =>
+        val allValidObligations = getObligations(bothQuarters, standardObligations, calendarObligations)
+
+        Seq(
+            TableRow(
+                content = Text(deadlineDate.toLongDateShort),
+                attributes = Map("id" -> s"quarterly-deadline-date-$index"),
+                classes = "govuk-!-font-weight-bold"
+            ),
+            TableRow(
+                content = Text(allValidObligations.headOption.map(obl => periodMessage(obl.obligation)).getOrElse("")),
+                attributes = Map("id" -> s"quarterly-period-$index")
+            ),
```

Comment:
so just to check if there are no valid obligations this will display and empty string in the Period column? 

Wondering if it would look odd probably more for content and design but imagine we may want something to say there is no deadline or the deadline is not valid etc. ?

Reply:
Think this is more of an oversight on my part. 

The old page for next updates actually omitts their table completely if there aren't any valid obligations. I'll get that implemented for this as well

Reply:
The period shouldn't ever be empty anyway as its a mandatory parameter. This is more of a safeguard more than anything which shouldn't ever be hit once i make the table only called if there are any deadlines.